bgpd/bgp_vty.c

```diff
@@ -9600,15 +9600,16 @@ int bgp_show_summary_vty(struct vty *vty, const char *name, afi_t afi,
 }
 
 /* `show [ip] bgp summary' commands. */
-DEFUN (show_ip_bgp_summary,
+DEFPY (show_ip_bgp_summary,
        show_ip_bgp_summary_cmd,
-       "show [ip] bgp [<view|vrf> VIEWVRFNAME] ["BGP_AFI_CMD_STR" ["BGP_SAFI_WITH_LABEL_CMD_STR"]] summary [established|failed] [json]",
+       "show [ip] bgp [<view|vrf> VIEWVRFNAME] ["BGP_AFI_CMD_STR" ["BGP_SAFI_WITH_LABEL_CMD_STR"]] [all$all] summary [established|failed] [json$uj]",
        SHOW_STR
        IP_STR
        BGP_STR
        BGP_INSTANCE_HELP_STR
        BGP_AFI_HELP_STR
        BGP_SAFI_WITH_LABEL_HELP_STR
+       "Display the entries for all address families\n"
```

Comment:
I think we already have the same for `show bgp summary`, right?

Reply:
Right, "show bgp summary" output is same as "show bgp all summary".
But, "show ip bgp summary" displays only ipv4 entries. 
So "show ip bgp summary" output is different from "show ip bgp all summary"

Reply:
"show ip bgp summary" displays only IPv4 entires as per design as IP in the show command means it is applicable only for IPV4 AFI extended that to dump all may not be right way to do this IMHO.

Reply:
Hi,
PR is opened for issue => https://github.com/FRRouting/frr/issues/6520
As you said, 
show [ip] bgp: 
"show bgp" commands display only ipv6(but by default unicast) entries &
"show ip bgp" commands display only ipv4 (but by default unicast )entries.
==> here "all" option is to display the entries for all AFIs/SAFIs (IPv4/6 unicast/multicast etc)

show [ip] bgp [ipv4|ipv6]:
show bgp ipv4 commands display only ipv4 (by default unicast) entries &
show ip bgp ipv6 commands display only ipv6 (by default unicast)
==> here "all" option is to display the entries for SAFIs (unicast, multicast)

Isn't it the requirement ?